libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/NavigationAppUserTurnstileEvent.kt

```diff
@@ -0,0 +1,25 @@
+package com.mapbox.navigation.core.telemetry.events
+
+import com.google.gson.Gson
+import com.mapbox.bindgen.Value
+import com.mapbox.common.TurnstileEvent
+import com.mapbox.navigation.base.metrics.MetricEvent
+import com.mapbox.navigation.base.metrics.NavigationMetrics
+import com.mapbox.navigation.metrics.internal.event.AppUserTurnstileInterface
+import org.json.JSONObject
+
+class NavigationAppUserTurnstileEvent(
+    override val event: TurnstileEvent
+) : MetricEvent, AppUserTurnstileInterface {
+
+    override val metricName: String
+        get() = NavigationMetrics.APP_USER_TURNSTILE
+
+    override fun toJson(gson: Gson): String = gson.toJson(this)
+
+    override fun toValue(): Value {
+        val jsonObject = JSONObject(toJson(Gson()))
+        val conversionResult = Value.fromJson(jsonObject.getJSONObject("event").toString())
```

Comment:
How do you know the key will be "event"? It's not specified anywhere in this class. I think you should refer to some constant here.

Reply:
@tarigo could you address this question? (I see you were working on this path). Can it be simplified to something like `TurnstileEvent.toValue(): Value`?

Reply:
cc @tatiana-yan

Reply:
Sorry, it was too long time ago and I lost the context. As far as I remember it was done to extract event and send it over common's sendTunstileEvent method.

Reply:
suppose I found it: `sendTurnstileEvent` is the separate method